### PR TITLE
adds a proc for when simplemobs attack themselves

### PR DESCRIPTION
**Pull request by ShadowDeath6**
fixes artificers also tell me if any other mob does this because I don't remember

:cl: ShadowDeath6
bugfix: Artificers can now heal themselves again.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 1]
```diff
@@ -59,17 +59,20 @@
...
-						   "<span class='cult'>You repair some of your own dents, leaving you at <b>[M.health]/[M.maxHealth]</b> health.</span>")
 		else
 			if(src != M)
 				M << "<span class='cult'>You cannot repair <b>[src]'s</b> dents, as it has none!</span>"
-			else
-				M << "<span class='cult'>You cannot repair your own dents, as you have none!</span>"
 	else if(src != M)
 		..()
 
+/mob/living/simple_animal/hostile/construct/AttackingSelf()
```
Comment (Super3222) [inline code/modules/mob/living/simple_animal/constructs.dm:68]: You don't need this...

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -652,9 +652,12 @@
 	if(harness && harness.on_melee_attack(src, T, 1))
 		return
 	if(T == src)
-		return
+		AttackingSelf()
 	T.attack_animal(src)
```
Comment (Super3222) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: They're going to attack themselves twice... why not just get rid of the check?

File: code/modules/mob/living/simple_animal/constructs.dm (hunk 1 above)
Comment (ShadowDeath6) [reply code/modules/mob/living/simple_animal/constructs.dm:68]: that's the entire point of this pr you fuk

Comment (Super3222) [reply code/modules/mob/living/simple_animal/constructs.dm:68]: You still don't need it.  You just need to remove the check in `AttackingTarget`.

Comment (ShadowDeath6) [reply code/modules/mob/living/simple_animal/constructs.dm:68]: I added the check for a reason super

Comment (Super3222) [reply code/modules/mob/living/simple_animal/constructs.dm:68]: So simple animals can't attack themselves? That shouldn't even be a feature. EVERYONE accidentally attacks themselves - it's just apart of the game. Exiling simple animals from it is snowflake-y.

Comment (ShadowDeath6) [reply code/modules/mob/living/simple_animal/constructs.dm:68]: Simple animals can only ever harm themselves if they click themselves. You can claim snowflake or hugbox but it's an irritating feature with literally no upsides and this PR adds a way for special effects to happen and fixes the problem.

**Comment by Super3222** [comment]
bad pr imo and we should get rid of that check

**Comment by ShadowDeath6** [comment]
if you can provide a real reason to convince me why we should allow simple animals to hit themselves other than "because other people can hit themselves" or "hugbox" I'll close the pr

**Comment by Time-Green** [comment]
I like it, alot of times I attack myself to see if there's something special

There isn't

**Comment by Super3222** [comment]
If you're making a proc for attacking themselves, at least make them attack themselves on regular. Artificers can do their heal thing on their own time.

**Pull request merged**